torch/distributed/distributed_c10d.py

```diff
@@ -436,6 +436,10 @@ def init_process_group(backend,
     _backend = _pg_map[_default_pg][0]
     _default_pg_init_method = init_method
 
+    # barrier at the end to ensure that once we return from this method, all
+    # process groups including global variables are updated correctly on all
+    # ranks.
+    barrier()
```

Comment:
We might need to guard this with try-except or if-else as it's possible that 3rd party extensions for c10d does not support barrier. See #45126

Reply:
I'm wondering how 3rd party extensions would work without supporting barrier? Currently ProcessGroup.hpp clearly defines the `barrier` method as pure virtual and as a result we should probably require extensions to support this. Do you know if there are any existing 3rd party extensions that do not support this? It looks like oneCCL and torch-ucc do support barrier().

Reply:
IIRC, we have one test c10d extension whose `barrier()` implementation just throws an unsupported error.